restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/AccountsEndpoint.java

```diff
@@ -802,6 +805,29 @@ private void validate(final MultivaluedMap<String, String> data) throws NullPoin
         } else if (!data.containsKey("Password")) {
             throw new NullPointerException("Password can not be null.");
         }
+
+        String emailAddress = data.getFirst("EmailAddress");
+        try {
+            InternetAddress emailAddr = new InternetAddress(emailAddress);
+            emailAddr.validate();
+        } catch (AddressException ex) {
+            String msg = String.format("Provided email address %s is not valid",emailAddress);
+            if (logger.isDebugEnabled()) {
+                logger.debug(msg);
+            }
+            throw new IllegalArgumentException(msg);
+        }
+
+        String clientLogin = data.getFirst("EmailAddress").split("@")[0];
+        for (char ch: ClientLoginConstrains.NOT_ALLOWED_CHARS) {
```

Comment:
can we move this code to ClientLoginCosntraints and reuse?=

Reply:
@jaimecasero Good suggestion, I will move it